File: server/datasets/python_source-code/Multiple-B2/organic_4.py
# Two Sum (hash map) - organic submission 4
# LeetCode-idiomatic class Solution style.

class Solution:
    def twoSum(self, nums, target):
        lookup = {}
        for index, value in enumerate(nums):
            need = target - value
            if need in lookup:
                return [lookup[need], index]
            lookup[value] = index
        return []


if __name__ == "__main__":
    sol = Solution()
    result = sol.twoSum([1, 5, 3, 8], 11)
    print("Indices:", result)

File: server/datasets/controlled_benchmarks/java/multiple_files/multiple_b2/organic_4.java
// Two Sum (hash map) - organic submission 4
// LeetCode-idiomatic class Solution style.

import java.util.Arrays;
import java.util.HashMap;
import java.util.Map;

class Solution {
    public int[] twoSum(int[] nums, int target) {
        Map<Integer, Integer> lookup = new HashMap<>();
        for (int index = 0; index < nums.length; index++) {
            int need = target - nums[index];
            if (lookup.containsKey(need)) {
                return new int[]{lookup.get(need), index};
            }
            lookup.put(nums[index], index);
        }
        return new int[]{};
    }

    public static void main(String[] args) {
        Solution sol = new Solution();
        int[] result = sol.twoSum(new int[]{1, 5, 3, 8}, 11);
        System.out.println("Indices: " + Arrays.toString(result));
    }
}
